homeassistant/components/shelly/entity.py

```diff
@@ -494,37 +487,26 @@ def entity_category(self) -> EntityCategory | None:
 class ShellyRestAttributeEntity(update_coordinator.CoordinatorEntity):
     """Class to load info from REST."""
 
+    entity_description: RestEntityDescription
+
     def __init__(
         self,
         wrapper: BlockDeviceWrapper,
         attribute: str,
-        description: RestAttributeDescription,
+        description: RestEntityDescription,
     ) -> None:
         """Initialize sensor."""
         super().__init__(wrapper)
         self.wrapper = wrapper
         self.attribute = attribute
-        self.description = description
-        self._name = get_block_entity_name(wrapper.device, None, self.description.name)
+        self.entity_description = description
+        self._attr_name = get_block_entity_name(wrapper.device, None, description.name)
+        self._attr_unique_id = f"{self.wrapper.mac}-{self.attribute}"
+        self._attr_device_info = DeviceInfo(
+            connections={(device_registry.CONNECTION_NETWORK_MAC, self.wrapper.mac)}
+        )
```

Comment:
This assumes that `wrapper.mac` is available right away.
However, I am unable to test this, so a code-owner should confirm.